### ISSUE TITLE
Build fails to find symbols if `--enable-lto` is used

### ISSUE THREAD

**Issue by Jayman2000** [FWGS/xash3d-fwgs#1300]
I’m running [NixOS](https://nixos.org/). When I try building Xash3D FWGS with `--enable-lto`, the compiler fails to find a bunch of symbols. It works fine if I don’t use `--enable-lto`.

[Build log](https://gist.github.com/Jayman2000/87c1865953e0d118027f770337eaf9d9)

**Comment by a1batross** [comment]
Fixed.

**Comment by Jayman2000** [comment]
I just tried to build Xash3D, and I ran into this same issue again: <https://gist.github.com/Jayman2000/1c5abe0a5a698003a963f02fc0888e78>.

**Comment by a1batross** [comment]
Yeah, so now it's an LTO problem. I didn't figured out why this happens, despite we copy all optimization flags.

**Comment by a1batross** [comment]
As a workaround, you can disable LTO. It doesn't gain very much.

Though it doesn't mean this won't get fixed.

**Comment by mittorn** [comment]
It seems to ignore static libraries after shared library. Need to debug linking order. Maybe,  some flags refers shared libraries and break static

**Comment by a1batross** [comment]
The linking order strikes back!

It's fun how some distros have a patch that reverses the order, and that's why it works for me, and some distros does not. 

The same bug also happens with Freedesktop SDK by the way.
